### PR TITLE
Fyst 580 1099 int implement xml builder for all states

### PR DESCRIPTION
**Pull request by tahsinaislam**
## Link to pivotal/JIRA issue
https://codeforamerica.atlassian.net/browse/FYST-548

## Is PM acceptance required?
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- Added in 1099-Int xml fields for the submission builder for states 

## How to test?
Unit test was created for the 1099-Int submission builder and personas were added for each state to test locally 

## Screenshots (for visual changes)
Screenshot 2024-10-24 at 1 53 29 PM

### CODE REVIEW

File: app/lib/pdf_filler/nc_d400_pdf.rb [hunk 1]
```diff
@@ -38,6 +38,7 @@ def hash_for_pdf
         y_d400wf_sv1yes: checkbox_value(@submission.data_source.spouse_veteran_yes?),
         y_d400wf_sv1no: checkbox_value(@submission.data_source.spouse_veteran_no?),
         y_d400wf_rs2yes: @submission.data_source.filing_status_mfj? ? 'Yes' : 'Off',
+        y_d400wf_fedex1no: 'Yes',
```
Comment (tahsinaislam) [inline app/lib/pdf_filler/nc_d400_pdf.rb:41]: added in a required field for nc d400 xml and pdf to make d400 xml stop erroring out.

File: app/lib/submission_builder/ty2024/states/nc/documents/d400.rb [hunk 2]
```diff
@@ -35,14 +36,14 @@ def document
                 # line 7 AdditionsToFAGI is blank
                 xml.FAGIPlusAdditions @submission.data_source.direct_file_data.fed_agi
                 xml.DeductionsFromFAGI calculated_fields.fetch(:NCD400_LINE_9)
-                xml.NumChildrenAllowed @submission.data_source.direct_file_data.qualifying_children_under_age_ssn_count
+                xml.NumChildrenAllowed @submission.data_source.direct_file_data.qualifying_children_under_age_ssn_count if @submission.data_source.direct_file_data.qualifying_children_under_age_ssn_count.present?
                 xml.ChildDeduction calculated_fields.fetch(:NCD400_LINE_10B)
```
Comment (tahsinaislam) [inline app/lib/submission_builder/ty2024/states/nc/documents/d400.rb:40]: other fixes on d400 to make it not have error messages

**Comment by powersurge360** [comment]
Wrong ticket appears to be linked, for convenience https://codeforamerica.atlassian.net/browse/FYST-580

File: app/lib/submission_builder/state1099_int.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  class State1099Int < SubmissionBuilder::Document
+    include SubmissionBuilder::FormattingMethods
+
+    def document
+      form1099int = @kwargs[:form1099int]
+      index = @kwargs[:index]
+      intake = @kwargs[:intake]
+
+      build_xml_doc("State1099Int", documentId: "State1099Int-#{index}") do |xml|
+        if form1099int.payer && form1099int.payer != ''
```
Comment (powersurge360) [inline app/lib/submission_builder/state1099_int.rb:11]: NIT: You may prefer `form1099int.payer.present?`

**Review by powersurge360 — APPROVED** [review]
Looks good. One minor nit that doesn't block anything.

**Pull request merged**